### PR TITLE
More Crowbar Acts

### PR DESCRIPTION
**Pull request by larentoun**
## What Does This PR Do

Converts some istype crowbars into crowbar acts

Left tool_behaviour check in some cases (such as multistep crafting)

## Why It's Good For The Game

~~I felt like it~~
Sounds for tool usage and more standartization for tool usage

## Testing

I clicked crowbar at things

### CODE REVIEW

**Comment by Henri215** [comment]
image

What exactly is the player facing change?

**Comment by larentoun** [comment]
> image
> 
> What exactly is the player facing change?

Uhh, added sounds to some things

**Comment by larentoun** [comment]
But changelog is mostly not needed it seems

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -369,10 +359,25 @@
 		return
 	return ..()
 
-/obj/machinery/suit_storage_unit/screwdriver_act(mob/user, obj/item/I)
-	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
+/obj/machinery/suit_storage_unit/crowbar_act(mob/living/user, obj/item/I)
+	if(!panel_open)
 		return
 	. = TRUE
+	if(!I.use_tool(src, user, volume = 50))
```
Comment (lewcc) [inline code/game/machinery/suit_storage_unit.dm:366]: Probably should keep the `I.tool_volume` call for the volume here

File: code/modules/research/server.dm [hunk 2]
```diff
@@ -142,15 +142,15 @@
 
 /obj/machinery/r_n_d/server/attackby(obj/item/O as obj, mob/user as mob, params)
 	if(exchange_parts(user, O))
-		return 1
+		return TRUE
+	. = ..()
```
Comment (lewcc) [inline code/modules/research/server.dm:146]:
```suggestion
	return ..()
```

File: code/modules/power/generators/portable generators/pacman.dm [hunk 3]
```diff
@@ -201,38 +201,43 @@
...
+	if(!active)
+		if(istype(O, /obj/item/storage/part_replacer) && panel_open)
 			exchange_parts(user, O)
-			return
-		else if(istype(O, /obj/item/crowbar) && panel_open)
-			default_deconstruction_crowbar(user, O)
-	else
-		return ..()
+		return
+	. = ..()
```
Comment (lewcc) [inline code/modules/power/generators/portable generators/pacman.dm:208]:
```suggestion
	return ..()
```

File: code/game/objects/structures/crates_lockers/largecrate.dm [hunk 4]
```diff
@@ -60,54 +62,60 @@
...
 
-/obj/structure/largecrate/lisa/attackby(obj/item/W as obj, mob/user as mob)	//ugly but oh well
-	if(istype(W, /obj/item/crowbar))
-		new /mob/living/simple_animal/pet/dog/corgi/Lisa(loc)
-	return ..()
+/obj/structure/largecrate/lisa/crowbar_act(mob/living/user, obj/item/I)
+	if(!I.use_tool(src, user, volume = 50))
+		return
+	new /mob/living/simple_animal/pet/dog/corgi/Lisa(loc)
+	. = ..()
```
Comment (lewcc) [inline code/game/objects/structures/crates_lockers/largecrate.dm:69]: change these all to `return ..()`

**Review by lewcc — COMMENTED** [review]
Code is mostly fine, just change the ending `. = ..()` to `return ..()` and please set the volumes to the `tool_volume`

File: code/game/objects/structures/janicart.dm [hunk 5]
```diff
@@ -87,32 +87,40 @@
...
 			mybag.attackby(I, user, params)
 	else
 		to_chat(usr, "<span class='warning'>You cannot interface your modules [src]!</span>")
 
+/obj/structure/janitorialcart/crowbar_act(mob/living/user, obj/item/I)
+	. = TRUE
+	user.visible_message("<span class='warning'>[user] begins to empty the contents of [src].</span>")
+	if(!I.use_tool(src, user, 3 SECONDS, I.tool_volume))
+		return
+	to_chat(usr, "<span class='notice'>You empty the contents of [src]'s bucket onto the floor.</span>")
```
Comment (Henri215) [inline code/game/objects/structures/janicart.dm:100]:
```suggestion
	to_chat(user, "<span class='notice'>You empty the contents of [src]'s bucket onto the floor.</span>")
```

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 6]
```diff
@@ -138,16 +119,58 @@
 	update_icon()
 	return
 
+/mob/living/simple_animal/bot/mulebot/crowbar_act(mob/living/user, obj/item/I)
+	if(!open || !cell)
+		return
+	. = TRUE
+	if(!I.use_tool(src, user, I.tool_volume))
+		return
+	cell.add_fingerprint(usr)
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:128]:
```suggestion
	cell.add_fingerprint(user)
```

**Review by Henri215 — APPROVED** [review]
Two minor things, looking good to mee

**Review by Burzah — APPROVED** [review]

**Pull request merged**